kernel/kernel-keymanager-service/src/main/java/io/mosip/kernel/clientcrypto/service/impl/ClientCryptoFacade.java

```diff
@@ -159,13 +159,23 @@ public byte[] encrypt(ClientType clientType, byte[] publicKey, byte[] dataToEncr
     public byte[] decrypt(byte[] dataToDecrypt) {
         byte[] encryptedSecretKey = Arrays.copyOfRange(dataToDecrypt, 0, symmetricKeyLength);
         byte[] secretKeyBytes =  Objects.requireNonNull(getClientSecurity()).asymmetricDecrypt(encryptedSecretKey);
-        byte[] iv = Arrays.copyOfRange(dataToDecrypt, symmetricKeyLength, symmetricKeyLength+ivLength);
-        byte[] aad = Arrays.copyOfRange(dataToDecrypt, symmetricKeyLength + ivLength, symmetricKeyLength+ivLength+aadLength);
-        byte[] cipher = Arrays.copyOfRange(dataToDecrypt, symmetricKeyLength + ivLength + aadLength,
-                dataToDecrypt.length);
-
         SecretKey secretKey = new SecretKeySpec(secretKeyBytes, "AES");
-        return cryptoCore.symmetricDecrypt(secretKey, cipher, iv, aad);
+
+        try {
+            byte[] iv = Arrays.copyOfRange(dataToDecrypt, symmetricKeyLength, symmetricKeyLength + ivLength);
+            byte[] aad = Arrays.copyOfRange(dataToDecrypt, symmetricKeyLength + ivLength, symmetricKeyLength + ivLength + aadLength);
+            byte[] cipher = Arrays.copyOfRange(dataToDecrypt, symmetricKeyLength + ivLength + aadLength,
+                    dataToDecrypt.length);
+            return cryptoCore.symmetricDecrypt(secretKey, cipher, iv, aad);
+        } catch (Throwable t) {
```

Comment:
Can we get a specific exception with a specific error before trying this workaround ?

Reply:
Because of array copy there is possibility of getting array index out of bound or any other run time error. so decided to catch Throwable.